src/package/docker.ts

```diff
@@ -0,0 +1,203 @@
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          incremental_key: incrementalKey,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          not_null: notNull,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          primary_key: primaryKey,
+          unique,
+        };
+      }),
+    };
+  });
+  await fs.writeFile(path.join(outputDirectory, 'tables.json'), JSON.stringify(tablesToWrite, null, 2));
+  logger.info(`Wrote ${tablesToWrite.length} tables to ${outputPath}`);
+};
+
+const runDockerCommand = async (logger: Logger, commandArguments: string[], cwd: string) => {
+  const process = execa('docker', commandArguments, {
+    cwd,
+    stdout: 'pipe',
+    stderr: 'pipe',
+  });
+  process.stdout?.on('data', (data) => {
+    logger.debug(data.toString());
+  });
+  process.stderr?.on('data', (data) => {
+    logger.debug(data.toString());
+  });
+  await process;
+};
+
+const writePackageJSON = async (
+  logger: Logger,
+  outputDirectory: string,
+  message: string,
+  pluginVersion: string,
+  supportedTargets: SupportedTarget[],
+  plugin: Plugin,
+) => {
+  const packageJsonPath = path.join(outputDirectory, 'package.json');
+  logger.info(`Writing package.json to ${packageJsonPath}`);
+  await fs.writeFile(
+    packageJsonPath,
+    JSON.stringify(
+      {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        schema_version: 1,
+        name: plugin.name(),
+        team: plugin.team(),
+        kind: plugin.kind(),
+        version: pluginVersion,
+        message: message,
+        protocols: [3],
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        supported_targets: supportedTargets,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        package_type: 'docker',
+      },
+      null,
+      2,
+    ),
+  );
+};
+
+const buildDockerfile = async (
+  logger: Logger,
+  outputDirectory: string,
+  pluginDirectory: string,
+  pluginVersion: string,
+  plugin: Plugin,
+) => {
+  const dockerfile = plugin.dockerFile();
+  const dockerFilePath = path.join(pluginDirectory, dockerfile);
+  if (!(await pathExists(dockerFilePath))) {
+    throw new Error(`docker file ${dockerFilePath} does not exist`);
+  }
+
+  logger.info(`Building ${plugin.buildTargets().length} targets`);
+  const supportedTargets = await pMap(
+    plugin.buildTargets(),
+    async ({ os, arch }) => {
+      const imageName = `cq-plugin-${plugin.name()}-${pluginVersion}-${os}-${arch}`;
+      const imageTar = `${imageName}.tar`;
+      const imagePath = `${outputDirectory}/${imageTar}`;
+      logger.info(`Building docker image ${imageName}`);
+      logger.debug(
+        `Running command 'docker buildx build -t ${imageName} --platform ${os}/${arch} -f ${dockerFilePath} .'`,
+      );
+      await runDockerCommand(
+        logger,
+        [
+          'buildx',
```

Comment:
We need `buildx` to cross platform images work inside GitHub Actions, we do the same in our current publishing pipeline https://github.com/cloudquery/cloudquery/blob/d91bda9afbd2e322beabc8a9bf34a2f0cd7407f5/.github/workflows/release_plugin.yml#L79